File: DataStructuresAndLibraries/pagina01/1023/1023.py
import math

t = 0
while True:
    try:
        n = int(input())

        if n == 0:
            break

        totalx, totaly = 0, 0
        consumos = []

        for _ in range(n):
            x, y = [int(el) for el in input().split(" ")]

            totalx += x
            totaly += y

            if y // x in consumos:
                consumos[y//x] += x
            else:
                consumos[y//x] = x


        consumo_total = ((100 * totaly) / totalx) / 100

        t += 1
        print(f"Cidade# {t}:")

        output = []
        keys = sorted(list(consumos.key()))
        for key in keys:
            output.append(f"{consumos[key]}-{key}")

        print(f"{" ".join(output)}")

        consumo_medio = math.floor(consumo_total)

        print(f"Consumo medio: {consumo_medio:.2f} m3.")


    except EOFError:
        break

File: DataStructuresAndLibraries/pagina01/1023/1023.ts
const fs = require('fs')
const input = fs.readFilSync('/dev/stdin', 'utf-8')
let lines = input.trim().split("\n")

lines.pop()

let contagemCidade = 0, count = 0
while(count < lines.length){

    let numeroDePessoas = parseInt(lines[count++])

    if(contagemCidade){ //adiciona uma linha em brtanco apenas depois se houver proximo cidade
        console.log('' )
    }

    //iniciando array de consumos com valores 0
    let consumos = new Array(201).fill(0)
    let [totalx, totaly] = [0, 0]

    for(let i = 0; i < numeroDePessoas; ++i){
        let [x, y] = lines[count++].split(' ').map(Number)

        if(x === 0) continue

        totalx += x
        totaly += y
        consumos[Math.floor(y / x)] += x
    }

    console.log(`Cidade# ${contagemCidade}:`)
    let output: string[] = []
    for(let i = 0; i < 201; ++i){
        if(consumos[i] > 0){
            output.push(`${consumos[i]}-${i}`)
        }
    }

    console.log(output.join(' '))

    if(totalx !== 0){
        let consumoMedio = Math.floor(100 * totaly / totalx) / 100
        console.log(`Consumo medio: ${consumoMedio.toFixed(2)} m3.`)
    }
}